Sources/Workspace/Workspace.swift

```diff
@@ -40,7 +40,15 @@ public enum WorkspaceError: Swift.Error {
     case incompatibleToolsVersion(rootPackage: AbsolutePath, required: ToolsVersion, current: ToolsVersion)
 
     /// The package at edit destination is not the one user is trying to edit.
-    case mismatchingDestinationPackage(path: AbsolutePath, destPackage: String, expectedPackage: String)
+    case mismatchingDestinationPackage(path: AbsolutePath, destinationPackage: String, expectedPackage: String)
+    
+    /// The dependency can't be pinned because it is being edited.
+    case cantPinDependencyInEditMode(dependency: ManagedDependency)
+}
+
+/// A warning in one of the workspace operations.
+public enum WorkspaceWarning {
```

Comment:
I agree. I only did it to mirror `WorkspaceError`. But in that case, I think we should do it for `WorkspaceError` too.

We'd have:

```
class Workspace {
    enum Error : Swift.Error {
        // ...
    }
    enum Warning {
        // ...
    }
    enum Info {
        // ...
    }
}
```